### PR TITLE
opencensus: add a sample for trace exemplar

### PR DESCRIPTION
**Pull request by dukeng**
Create a sample code that will be referenced in the documentation for Trace Exemplar instrumentation.

### CODE REVIEW

**Comment by dukeng** [comment]
@tbpg Please review

**Comment by dukeng** [comment]
I cannot add reviewers to this PR. @anguillanneuf or @tbpg Could you please take a look at this PR and/or let me know how to formally request a review?

File: opencensus/trace_exemplar_quickstart/main.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+				Exemplars: []*distributionpb.Distribution_Exemplar{
+					&distributionpb.Distribution_Exemplar{Value: 1, Timestamp: &googlepb.Timestamp{Seconds: end - 30}},
+					&distributionpb.Distribution_Exemplar{Value: 4, Timestamp: &googlepb.Timestamp{Seconds: end - 30}},
+				},
+			},
+		}},
+	}
+	// [END monitoring_opencensus_configure_trace_exemplar]
+
+	// Writes time series data.
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:75]: Is the rest of this code needed for the sample?

File: opencensus/trace_exemplar_quickstart/main.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	googlepb "github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+	distributionpb "google.golang.org/genproto/googleapis/api/distribution"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+func main() {
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:39]: Please add a test.

File: opencensus/trace_exemplar_quickstart/main.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:1]: Please move this to the `opencensus` directory directly. Not a quickstart anymore, so it doesn't need to be in another directory.

File: opencensus/trace_exemplar_quickstart/main.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample opencensus_spanner_quickstart contains a sample application that
+// uses Google Spanner Go client, and reports metrics
+// and traces for the outgoing requests.
+package trace_exemplar_quickstart
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:18]: Update with the new package name (can do `opencensus` or `samples` or something) based on the new directory.

File: opencensus/trace_exemplar_quickstart/main.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				},
+			},
+		}},
+	}
+	return dataPoint
+}
+
+// [END monitoring_opencensus_configure_trace_exemplar]
+
+func writeTimeSeriesData(w io.Writer, projectID string) error {
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:63]: Can inline this in the test since it's not used in the sample.

File: opencensus/trace_exemplar_quickstart/main_test.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestWriteTimeSeriesData(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	buf := &bytes.Buffer{}
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main_test.go:27]: No need to write to a buffer once you inline `writeTimeSeriesData` here. Can just test there isn't an error.

File: opencensus/trace_exemplar_quickstart/main.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	ctx := context.Background()
+	dataPoint := createDataPointWithExemplar()
+	// Creates a client.
+	client, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "failed to create client"))
+	}
+	// Writes time series data.
+	if err := client.CreateTimeSeries(ctx, &monitoringpb.CreateTimeSeriesRequest{
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:72]: A little hard to follow this indentation with the inline `err` and `CreateTimeSeriesRequest` declarations. Can declare `req` separately then this if will be easier to follow.

File: opencensus/trace_exemplar_quickstart/main.go [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+// [END monitoring_opencensus_configure_trace_exemplar]
+
+func writeTimeSeriesData(w io.Writer, projectID string) error {
+	ctx := context.Background()
+	dataPoint := createDataPointWithExemplar()
+	// Creates a client.
+	client, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "failed to create client"))
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:69]: Will be a test failure when it's moved there, but return errors instead of calling `log.Fatal`.

File: opencensus/trace_exemplar_quickstart/main.go [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+			ValueType:  metricpb.MetricDescriptor_DISTRIBUTION,
+			Resource: &monitoredrespb.MonitoredResource{
+				Type:   "generic_node",
+				Labels: map[string]string{"location": "us-east1-a", "namespace": "space", "node_id": "1"},
+			},
+			Points: []*monitoringpb.Point{dataPoint},
+		}},
+	}); err != nil {
+		return err
+	} else {
```
Comment (tbpg) [inline opencensus/trace_exemplar_quickstart/main.go:86]: No need for the else if the `!= nil` check returns early. 

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

File: opencensus/trace_exemplar_quickstart/main.go (hunk 5 above)
Comment (dukeng) [reply opencensus/trace_exemplar_quickstart/main.go]: Moved to the test file.

File: opencensus/trace_exemplar_quickstart/main.go (hunk 3 above)
Comment (dukeng) [reply opencensus/trace_exemplar_quickstart/main.go]: Done

File: opencensus/trace_exemplar_quickstart/main.go (hunk 8 above)
Comment (dukeng) [reply opencensus/trace_exemplar_quickstart/main.go]: Done

File: opencensus/trace_exemplar_quickstart/main.go (hunk 7 above)
Comment (dukeng) [reply opencensus/trace_exemplar_quickstart/main.go]: Done. I moved err out of the if statement

File: opencensus/trace_exemplar_quickstart/main.go (hunk 9 above)
Comment (dukeng) [reply opencensus/trace_exemplar_quickstart/main.go]: Omitted the nil check
